WPThemeReview/Sniffs/ThouShallNotUse/ReservedTemplateNameSniff.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * WPThemeReview Coding Standard.
+ *
+ * @package WPTRT\WPThemeReview
+ * @link    https://github.com/WPTRT/WPThemeReview
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WPThemeReview\Sniffs\ThouShallNotUse;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Check if the template file is using reserved file name as a part of their name.
+ *
+ * @since 0.2.0
+ */
+class ReservedTemplateNameSniff implements Sniff {
+
+	/**
+	 * Error message template.
+	 *
+	 * @var string
+	 */
+	const ERROR_MSG = 'File template should not use a reserved name as a prefix. Found: "%s".';
```

Comment:
From a dev-education point of view, I think this message could be improved as it doesn't explain _why_.

Reply:
Would this be a better explanation

```
'Template file should not use a reserved name as a prefix, because it will be recognised by WordPress and will apply to one specific page only. Found: "%s".'
```

?

Reply:
Still not sure if that's clear enough. Maybe ask some theme reviewers for some suggestions ?

Rough draft from my end:
`Template files should not be slug specific. The file name used for this template will be interpreted by WP as %s-$slug and only applied when the %s named $slug will be loaded.`

Reply:
I thought it has to be something short 😆 Your suggestion looks great 😄 I'll ask in the channel :+1:

Reply:
The message can be as long or short. Most importantly, it has to be informative. You can even use new lines in error messages if it helps (though it may annoy people with very wide screens).

Errorcodes should be relatively short and modular, i.e. make it easy for people to ignore just one type of issue.

Reply:
I've asked and this seems to be better:

`
Template files should not be slug specific. The file name used for this template will be interpreted by WP as %1$s-%2$s and only applied when a %1$s with the slug %2$s is loaded.
`

I'm also capturing the `slug` and `prefix` so the error message would be like this 

```
FILE: /Users/infinum-denis/Desktop/TEST/page-comments.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 1 | ERROR | Template files should not be slug specific. The file
   |       | name used for this template will be interpreted by WP as
   |       | page-comments and only applied when a page with the
   |       | slug comments is loaded.
----------------------------------------------------------------------
```